pandas/core/indexing.py

```diff
@@ -2167,8 +2169,10 @@ def check_bool_indexer(index: Index, key) -> np.ndarray:
         result = np.asarray(result, dtype=bool)
         result = check_array_indexer(index, result)
     else:
```

Comment:
can you make this an elif, then have a separate else clause.

put comments on what each is catching